File: find_a_peak.java
// EGY (nem a legelső) csúcs keresése
public static int find_a_peak(int[] a) throws IllegalArgumentException
{
  if (a == null) {
    throw new IllegalArgumentException();
  }

  if (a.length == 0) {
    return null;
  }
  if (a.length == 1) {
    return 0;
  }
  if (a[0] >= a[1]) {
    return 0;
  }
  if (a[a.length - 1] >= a[a.length - 2]) {
    return a.length - 1;
  }

  for (int j = 2; j < a.length - 1; j++) {
    if (a[j] >= a[j-1] && a[j] >= a[j+1]) {
      return j;
    }
  }

  return null;
}

File: find_a_peak.js
// EGY (nem a legelső) csúcs keresése
function find_a_peak(a) {
  if (a.length === 0) {
    return null;
  }
  if (a.length === 1) {
    return 0;
  }
  if (a[0] >= a[1]) {
    return 0;
  }
  if (a[a.length - 1] >= a[a.length - 2]) {
    return a.length - 1;
  }

  for (let i = 2; i < a.length - 1; i++) {
    if (a[i] >= a[i - 1] && a[i] >= a[i + 1]) {
      return i;
    }
  }

  return null;
}

let array = [1, 3, 4, 3, 5, 1 /*, 3*/];
const egyszeruCsucs = find_a_peak(array);
console.log(egyszeruCsucs);
